text/0000-bots-for-health.md

```diff
@@ -0,0 +1,93 @@
+*   Start date: 2020-11-04
+*   Scope: Collective
+*   RFC PR: <!-- leave this empty -->
+*   Implementation issue(s): <!-- leave this empty -->
+
+# Summary
+
+It’s becoming harder to handle the incoming stream of questions and issues, so I
+thought I’d open a discussion to talk about how to tackle that.
+
+Bots may help.
+
+This is a conversation starter: **I’m curious to hear what y’all think would
+(not) be useful**.
+
+## Motivation
+
+It’s becoming harder to handle the incoming stream of questions and issues, so I
+thought I’d open a discussion to talk about how to tackle that.
+
+I’m assuming this is coming from a couple of places:
+
+*   unified is becoming more popular
+*   MDX, recently react-markdown, and potentially more in the future, joined
+*   Discussions is an easy way for folks to ask questions
+
+Regardless of why, a different type of user is reaching unified (an exemplary
+question yesterday was: “why are there margins on my `<p>`”,
+[remarkjs/react-markdown#503](https://github.com/remarkjs/react-markdown/issues/503)).
+A lot of these questions are XY problems and result in maintainers acting like
+a rubber duck for three back-and-forths.
+
+I hope that automation can help deal with the flow of incoming questions so that
+maintainers can stay healthy.
+
+## Detailed design
+
+For a solution, here are a couple broad ideas:
+
+*   Improve docs on unifiedjs.com (important, but to be discussed somewhere
+    else)
+*   Improve contributing, support, and issue templates
+*   Shared saved responses (see
+    <https://github.com/electron/governance/tree/master/playbooks/responses>
+    and
+    <https://github.com/electron/electron/issues?q=is%3Aissue+is%3Aclosed>)
+*   Add automation/bots
+
+Specifically, for automation, I’m thinking:
+
+*   Enforce filled out issue/pr/discussion templates
+*   “Lint” issue checklists: if someone is on an old Node, respond and close
+    that those aren’t supported; or enforce that
+    `* [ ] I promise to adhere to the code of conduct` is checked
+*   Automate [label](https://github.com/unifiedjs/github-tools/labels)/issue
+    interplay:
+    if maintainer adds `type/question`, the issue is closed and a predefined
+    response posted to ask in discussions instead;
+    `platform/*` can be found from the issue checklist;
+    merging can add `status/merge`, or `status/wontfix` can close one;
+    `area/*` could be added to the issue title?
+*   Maybe difficult, but: GH has support for finding similar issues, but doesn’t
+    scan all our repos (or discussions), maybe we can do that ourselves: respond
+    on an issue with similar issues?
+*   Lock old issues
+
+I’m thinking this can be done in GH actions, which seems to be pretty fast.
+That does mean the action must be added to all repos, but then it’s also a nice
+opt-in.
```

Comment:
There are some options.
Many GH automations are built with https://probot.github.io
Which offers some flexibility on where/how it is hosted.
We could make these GitHub actions, we could also make this a GitHub app.

Reply:
Actions seems to be relatively nice for this, because:
- It’s fast
- It’s automatically authentication to change labels, post comments, etc
- GHs limits seem to be based per repo, of which we have a lot, but they each have low activity, so it seems we won’t reach thresholds.

See also:
- Probot vs. Actions: https://jasonet.co/posts/probot-app-or-github-action-v2
- Actions vs. Apps: https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/about-actions#comparing-github-actions-to-github-apps